### ISSUE TITLE
Delta filter

### ISSUE THREAD

**Issue by b3nn0** [dbuezas/lovelace-plotly-graph-card#173]
**Describe the solution you'd like**

It would be cool if we were able to specify a filter like
`derivate: singlevalue` or `derivate: 1` or something,
This would just compare a value with it's predecessor and yield the difference.

Usecase:
With the recent statistics feature, we already get nicely resampled data.
By using `period: auto`, the resampling is actually dynamic. If you zoom in, you get 5 minute resampling, if you zoom out, you get 1 month resampling, etc.
By specifying `derivate: 1`, the derivate would therefore be nicely scaling as well.
Example by using a lambda: a dynamic gas consumption bar chart, which shows a "consumption per 5 minutes" graph when zooming in, or a "consumption per month" graph when zooming out - similar to the stock Energy dashboard, where one can also select the desired time range.

Zoom in: hourly gas consumption:
![grafik](https://user-images.githubusercontent.com/1858945/211150461-d5128c90-1db6-475b-b62f-9f2f98ed883f.png)

Zoom out: daily gas consumption:
![grafik](https://user-images.githubusercontent.com/1858945/211150480-7750b231-3ca6-4d84-8db9-3b83fa178358.png)

Lambda to produce those:
```
    period: auto
    lambda: |-
      (ys, xs) => {
        const res = []
        for (let i = 1; i < ys.length; i++) {
          res.push((ys[i] - ys[i-1]))
        }
        xs.shift()
        return {x: xs, y: res }
      }
```

Implementing this should thus be trivial. It's just a value-to-value-derivate, as the resampling is already done by the statistics.

Alternatively, this could also be a separate filter instead of re-using the derivate filter for it. Not sure how to call it though..

**How would it be defined in yaml?**
```yaml
type: custom:plotly-graph
entities:
  - entity: sensor.gasverbrauch_heizung
  - filters:
    - derivate: 1
```

**Comment by dbuezas** [comment]
Got it. What do you think about a new filter instead? I think "differential" would be the right name.

In the meantime, (as a workaround) it should be somewhat equivalent to:
```yaml
- resample: 1h
- derivate: h
```

Because:
* The formula in derivative is `derivated_y = delta_y / delta_x * unit`
* Resample ensures `delta_x = 1h`
* The `unit` is set to `1h`
So it ends up returning `delta_y`

**Comment by dbuezas** [comment]
The new filter could be called:
* differential
* differentials
* deltas
* y_deltas
* <your suggestion?>

What do you think would be most intuitive?

**Comment by dbuezas** [comment]
Oh, right it breaks with auto period. 
I like delta_y. I'll add it to the next release 👍

**Comment by dbuezas** [comment]
In the end I went for `delta`.
Will be out in the next release, but if you want to review the PR or know how to, you can install that branch.

**Comment by b3nn0** [comment]
Awesome :-)
Unfortunately I don't know, so I'll wait until it's in master

**Comment by dbuezas** [comment]
I'll wait a couple of days for the next release to gather a couple more changes and you should be good to go

**Comment by dbuezas** [comment]
Closed by the release of v3.0.0
